code/socialDistribution/views.py

```diff
@@ -436,65 +436,62 @@ def share_post(request, id):
     return redirect('socialDistribution:home')
 
 
-def edit_post(request, id):
+def edit_post(request, id, post_host):
     """
         Edits an existing post
     """
-    author = LocalAuthor.objects.get(user=request.user)
-    post = LocalPost.objects.get(id=id)
+    if post_host == 'remote':
+        post = get_object_or_404(InboxPost, id=id)
+        request_url = post.author.strip('/') + f'/posts/{id}'
+    else:
+        post = get_object_or_404(LocalPost, id=id)
+        host = request.get_host()
+        request_url = f'http://{host}/{API_PREFIX}/author/{post.author.id}/posts/{id}'
+        
```

Comment:
I think this may be more complicated than necessary. We won't ever have to edit a remote post, right? Since the author will only have posts on the local server.

Reply:
Ohh yeah that makes sense. So we'll edit the LocalPost and then we'll need to update the InboxPosts that are created after dispatching

Reply:
Ya exactly. We don't really do anything to the LocalPost that was updated. We just need each InboxPost to refresh by checking its url_id for updates every now and then.